### PR TITLE
Add documentation for adaptive scaling

### PR DESCRIPTION
**Pull request by mrocklin**
cc @jacobtomlinson @timkock , your review on this would be welcome

### CODE REVIEW

**Comment by jacobtomlinson** [comment]
Looks good to me!

File: docs/source/setup/adaptive.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+Most Dask deployments are static, with a single scheduler and a fixed number of
+workers.  This results in predictable behavior, but is wasteful of resources in
+two situations:
+
+1.  The user may not be using the cluster, perhaps they are busy
+    interpretting a recent result or plot, and so the workers sit idly,
+    taking up valuable shared resources from other potential users
+2.  The user may be very active, and is limited by their original allocation.
+
+Particularlly efficient users may learn to manually add and remove workers
```
Comment (jakirkham) [inline docs/source/setup/adaptive.rst]: `Particularlly` -> `Particularly`

File: docs/source/setup/adaptive.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+
+Scaling Heuristics
+------------------
+
+The Dask scheduler tracks a variety of information that is useful to correctly
+allocate the number of workers:
+
+1.  The historical runtime of every function and task that it has seen,
+    and all of the functions that it is currently able to run for users
```
Comment (jakirkham) [inline docs/source/setup/adaptive.rst]: Out of curiosity, how is this stored?

**Comment by mrocklin** [comment]
Thanks.  Fixed typos.  

We store task durations in exponentially weighted moving averages in `Scheduler.task_duration`: 
https://github.com/dask/distributed/blob/master/distributed/scheduler.py#L2093-L2103

**Comment by mrocklin** [comment]
Merging soon if there are no further comments

**Comment by jakirkham** [comment]
Thanks @mrocklin. :)

**Pull request merged**